src/uagents/agent.py

```diff
@@ -232,12 +286,46 @@ def _schedule_registration(self):
 
         return (expiry - height) * BLOCK_INTERVAL
 
+    def _is_almanac_registered(self) -> bool:
+        query_msg = {"query_records": {"agent_address": self.address}}
+        response = self._reg_contract.query(query_msg)
+
+        if not response["record"]:
+            return False
+        return True
+
     def get_registration_sequence(self) -> int:
         query_msg = {"query_sequence": {"agent_address": self.address}}
         sequence = self._reg_contract.query(query_msg)["sequence"]
 
         return sequence
 
+    def get_agent_address(self, name: str) -> str:
+        query_msg = {"domain_record": {"domain": f"{name}.agent"}}
+        res = self._service_contract.query(query_msg)
+        if res["record"] is not None:
+            registered_address = res["record"]["records"][0]["agent_address"]["records"]
+            if len(registered_address) > 0:
+                return registered_address[0]["address"]
+            return 0
+        return 1
+
+    def is_name_available(self, name: str):
+        query_msg = {"domain_record": {"domain": f"{name}.agent"}}
+        return self._service_contract.query(query_msg)["is_available"]
+
+    def is_owner(self, name: str):
+        query_msg = {
+            "permissions": {
+                "domain": f"{name}.agent",
+                "owner": {"address": {"address": str(self.wallet.address())}},
+            }
+        }
+        permission = self._service_contract.query(query_msg)["permissions"]
+        if permission == "admin":
+            return True
+        return False
```

Comment:
It might make sense to have contract classes that inherit from `LedgerContract` for both the `almanac` and `name_service` contracts. Then we could add these methods to it like `name_service.is_owner(name: str, address: str)`.